flink-table/flink-table-storage/src/main/java/org/apache/flink/table/storage/logstore/LogSink.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.storage.logstore;
+
+import org.apache.flink.api.connector.sink.Committer;
+import org.apache.flink.api.connector.sink.Sink;
+import org.apache.flink.api.connector.sink.Sink.InitContext;
+import org.apache.flink.api.connector.sink.SinkWriter;
+import org.apache.flink.core.io.SimpleVersionedSerializer;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * A {@link Sink} for log system.
+ *
+ * <p>Add this specialized interface for various reasons:
+ *
+ * <ul>
+ *   <li>It offers additional {@link WriteCallback} to {@link #createWriter}.
+ *   <li>It does have local committer.
+ *   <li>It does not have global committer.
+ * </ul>
+ */
+public interface LogSink<CommT, WriterStateT> extends Serializable {
```

Comment:
Let's try our best to avoid a second Sink interface. It might be easiest to simply add the Callback to InitContext.

Reply:
But this must add a public api to `InitContext` or a subclass.

Reply:
If you expect connectors to implement it, it needs to be public API. All connectors will be eventually removed from the main repo.

Reply:
I'd still prefer to just spawn a secondary sink and treat it as an implementation detail.